pkg/secrets/config/config.go

```diff
@@ -28,63 +29,266 @@ type Config struct {
+			k8sSecretsArr = k8sSecretsArr[:len(k8sSecretsArr)-1]
+		} else {
+			k8sSecretsStr = settings["K8S_SECRETS"]
+			k8sSecretsStr = strings.ReplaceAll(k8sSecretsStr, " ", "")
+			k8sSecretsArr = strings.Split(k8sSecretsStr, ",")
+		}
+	}
+
+	retryCountLimitStr := settings["conjur.org/retry-count-limit"]
+	if retryCountLimitStr == "" {
+		retryCountLimitStr = settings["RETRY_COUNT_LIMIT"]
+	}
+	retryCountLimit := parseIntFromStringOrDefault(retryCountLimitStr, DEFAULT_RETRY_COUNT_LIMIT, MIN_RETRY_VALUE)
+
+	retryIntervalSecStr := settings["conjur.org/retry-interval-sec"]
+	if retryIntervalSecStr == "" {
+		retryIntervalSecStr = settings["RETRY_INTERVAL_SEC"]
+	}
+	retryIntervalSec := parseIntFromStringOrDefault(retryIntervalSecStr, DEFAULT_RETRY_INTERVAL_SEC, MIN_RETRY_VALUE)
 
 	return &Config{
 		PodNamespace:       podNamespace,
-		RequiredK8sSecrets: requiredK8sSecrets,
+		RequiredK8sSecrets: k8sSecretsArr,
 		RetryCountLimit:    retryCountLimit,
 		RetryIntervalSec:   retryIntervalSec,
 		StoreType:          storeType,
-	}, nil
+	}
+}
+
+func validateAnnotationKey(key string) (string, map[string][]string, error) {
+	// Validate that a given annotation key is formatted as "conjur.org/xyz"
+	// Record Info level log if a key conforms to the formatting standard but
+	// is not recognized as either a Secrets Provider config or Push to File config annotation
+	//
+	// If the annotation is for Push to File config, the ValidAnnotations function
+	// needs to be aware of the annotation's valid prefix in order to perform input validation,
+	// so this function returns:
+	//   - either the key, or the key's valid prefix
+	//   - the Map in which the key or prefix was found
+	//   - the success status of the operation
+	if !strings.HasPrefix(key, "conjur.org/") {
+		return "", nil, fmt.Errorf(messages.CSPFK011I, key)
+	}
+
+	if _, ok := secretsProviderAnnotations[key]; ok {
+		return key, secretsProviderAnnotations, nil
+	} else if prefix, ok := valuePrefixInMapKeys(key, pushToFileAnnotationPrefixes); ok {
+		return prefix, pushToFileAnnotationPrefixes, nil
+	} else {
+		return "", nil, fmt.Errorf(messages.CSPFK012I, key)
+	}
 }
 
-func parseIntFromEnvOrDefault(environmentVariable string, defaultValue int, minValue int) int {
-	envString := os.Getenv(environmentVariable)
-	envValueInt, err := strconv.Atoi(envString)
-	if err != nil || envValueInt < minValue {
+func validateAnnotationValue(key string, value string, acceptedValueInfo []string) error {
+	// given a key/value pair, where the key is confirmed to be a Secrets Provider config annotation
+	// validated that the value is of valid type, or confirm that the value is
+	// in the range of enumerated and acceptable values
+	switch targetType := acceptedValueInfo[0]; targetType {
+	case "int":
+		_, err := strconv.Atoi(value)
+		if err != nil {
+			return fmt.Errorf(messages.CSPFK042E, key, value, "Integer")
+		}
+	case "bool":
+		if value != "true" && value != "false" {
```

Comment:
No, not with this implementation. If we want, I can use `strconv.ParseBool` to determine if a boolean setting was acceptable, which would allow "TRUE" and "FALSE", and some others:

> ParseBool returns the boolean value represented by the string. It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False. Any other value returns an error.